File: java/javasearch/src/test/java/javasearch/SearchSettingsTest.java
package javasearch;

import org.junit.Test;

import java.util.Set;
import java.util.regex.Pattern;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class SearchSettingsTest {

    public SearchSettingsTest() {

    }

    @Test
    public final void testDefaultSearchSettings() {
        SearchSettings settings = new SearchSettings();
        assertEquals(DefaultSearchSettings.COLORIZE, settings.getColorize());
        assertEquals(DefaultSearchSettings.FIRST_MATCH, settings.getFirstMatch());
        assertEquals(DefaultSearchSettings.LINES_AFTER, settings.getLinesAfter());
        assertEquals(DefaultSearchSettings.LINES_BEFORE, settings.getLinesBefore());
        assertEquals(DefaultSearchSettings.MAX_LINE_LENGTH, settings.getMaxLineLength());
        assertEquals(DefaultSearchSettings.MULTI_LINE_SEARCH, settings.getMultiLineSearch());
        assertEquals(DefaultSearchSettings.PRINT_LINES, settings.getPrintLines());
        assertEquals(DefaultSearchSettings.PRINT_RESULTS, settings.getPrintResults());
        assertEquals(DefaultSearchSettings.SEARCH_ARCHIVES, settings.getSearchArchives());
        assertEquals(DefaultSearchSettings.UNIQUE_LINES, settings.getUniqueLines());
    }

    @Test
    public final void testAddExtensions() {
        SearchSettings settings = new SearchSettings();
        settings.addInExtension("java,scala");
        Set<String> inExtensions = settings.getInExtensions();
        assertEquals(inExtensions.size(), 2);
        assertTrue(inExtensions.contains("java"));
        assertTrue(inExtensions.contains("scala"));
    }

    @Test
    public final void testAddPattern() {
        SearchSettings settings = new SearchSettings();
        settings.addSearchPattern("Searcher");
        Set<Pattern> searchPatterns = settings.getSearchPatterns();
        assertEquals(searchPatterns.size(), 1);
    }

    @Test
    public final void testSetArchivesOnly() {
        SearchSettings settings = new SearchSettings();
        settings.setArchivesOnly(true);
        assertTrue(settings.getArchivesOnly());
        assertTrue(settings.getSearchArchives());
    }

    @Test
    public final void testSetDebug() {
        SearchSettings settings = new SearchSettings();
        settings.setDebug(true);
        assertTrue(settings.getDebug());
        assertTrue(settings.getVerbose());
    }
}

File: go/gosearch/pkg/gosearch/searchsettings_test.go
package gosearch

import "testing"

func TestDefaultSearchSettings(t *testing.T) {
	settings := GetDefaultSearchSettings()
	if settings.ArchivesOnly() ||
		settings.Debug() ||
		settings.FirstMatch() ||
		settings.FollowSymlinks() ||
		settings.IncludeHidden() ||
		settings.MultiLineSearch() ||
		settings.PrintDirs() ||
		settings.PrintFiles() ||
		settings.PrintLines() ||
		settings.PrintMatches() ||
		settings.PrintResults() ||
		settings.PrintUsage() ||
		settings.PrintVersion() ||
		!settings.Recursive() ||
		settings.SearchArchives() ||
		settings.UniqueLines() ||
		settings.Verbose() {
		t.Errorf("settings did not match defaults")
	}
}

func TestAddPattern(t *testing.T) {
	settings := GetDefaultSearchSettings()
	settings.AddSearchPattern("Searcher")
	if settings.SearchPatterns().IsEmpty() {
		t.Errorf("SearchPatterns should not be empty")
	}
}

func TestAddExtensions(t *testing.T) {
	settings := GetDefaultSearchSettings()
	settings.AddInExtension("go,hs")
	if len(settings.InExtensions()) != 2 {
		t.Errorf("InExtensions should have two elements")
	}
}

func TestSetArchivesOnly(t *testing.T) {
	settings := GetDefaultSearchSettings()
	settings.SetArchivesOnly(true)
	if !settings.ArchivesOnly() {
		t.Errorf("ArchivesOnly should be true")
	}
	if !settings.SearchArchives() {
		t.Errorf("SearchArchives should be true")
	}
}
